src/myst/interpreter/native_lib/time.cr

```diff
@@ -0,0 +1,49 @@
+module Myst
+  class Interpreter
+    NativeLib.method :time_now, Value do
+      t = Time.now
+      
+      instance = NativeLib.instantiate(self, this.as(TType), [
+        TInteger.new(t.year.to_i64), 
+        TInteger.new(t.month.to_i64), 
+        TInteger.new(t.day.to_i64), 
+        TInteger.new(t.hour.to_i64), 
+        TInteger.new(t.minute.to_i64), 
+        TInteger.new(t.second.to_i64)
+      ] of Value)
+
+      instance
+    end
+
+    NativeLib.method :time_to_s, TInstance, format : TString? do
```

Comment:
I hadn't thought of using a nilable type for the expected argument. Makes a lot of sense now that I see it in practice.